### PR TITLE
docs: add v18 migration guide

### LINKED ISSUE #4376: Add v18 migration guide
### Information

Add a migration guide for V18.

### Documentation page

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #4376

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by markostanimirovic — APPROVED** [review]

File: projects/ngrx.io/content/guide/migration/v18.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+- RxJS version ^6.5.x || ^7.5.0
+
+## Breaking changes
+
+### @ngrx/store
+
+#### Merge `Action` and `TypedAction` interfaces
+
+The Action and TypedAction interfaces are merged into one interface.
+This change has a ng-update schematic that will automatically update your code.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/migration/v18.md:29]:
```suggestion
This change has an ng-update schematic that automatically updates your code.
```

File: projects/ngrx.io/content/guide/migration/v18.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+The Action interface accepts a generic type parameter that represents the payload type (defaults to string).
+The TypedAction interface is removed.
+
+### @ngrx/effects
+
+#### Remove `concatLatestFrom` operator
+
+The `concatLatestFrom` operator has been removed from `@ngrx/effects` in favor of the `@ngrx/operators` package.
+This change has a ng-update schematic that will automatically update your code.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/migration/v18.md:45]:
```suggestion
This change has an ng-update schematic that automatically updates your code.
```

File: projects/ngrx.io/content/guide/migration/v18.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+```ts
+import { concatLatestFrom } from '@ngrx/operators';
+```
+
+### @ngrx/component-store
+
+#### Remove `tapResponse` operator
+
+The `tapResponse` operator has been removed from `@ngrx/component-store` in favor of the `@ngrx/operators` package.
+This change has a ng-update schematic that will automatically update your code.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/migration/v18.md:64]:
```suggestion
This change has an ng-update schematic that automatically updates your code.
```

File: projects/ngrx.io/content/guide/migration/v18.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+ng update @ngrx/store@18
+```
+
+## Dependencies
+
+Version 18 has the minimum version requirements:
+
+- Angular version 18.x
+- Angular CLI version 18.x
+- TypeScript version 5.x
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/migration/v18.md:19]:
```suggestion
- TypeScript version 5.4.x
```

File: projects/ngrx.io/content/guide/migration/v18.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
+# V18 Update Guide
+
+## Angular CLI update
+
+NgRx supports using the Angular CLI `ng update` command to update your dependencies. Migration schematics are run to make the upgrade smoother. These schematics will fix some of the breaking changes.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/migration/v18.md:5]:
```suggestion
NgRx supports using the Angular CLI `ng update` command to update your dependencies. Migration schematics are run to make the upgrade smoother. These schematics fix some of the breaking changes.
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Comment by timdeschryver** [comment]
Thanks @brandonroberts !

**Review by brandonroberts — APPROVED** [review]
🚀

**Pull request merged**